examples/dinput8_hook_d3d12/src/lib.rs

```diff
@@ -0,0 +1,114 @@
+#![feature(once_cell)]
+#![allow(non_snake_case, unused_variables, unreachable_patterns)]
+
+use std::ffi::c_void;
+use std::sync::LazyLock;
+use std::thread;
+
+use hudhook::hooks::dx12::ImguiDx12Hooks;
+use hudhook::hooks::{self, ImguiRenderLoop, ImguiRenderLoopFlags};
+use hudhook::log::trace;
+use hudhook::reexports::DLL_PROCESS_ATTACH;
+use hudhook::renderers::imgui_dx12::imgui;
+use imgui::Condition;
+use simplelog::{Color, ColorChoice, ConfigBuilder, Level, LevelFilter, TermLogger, TerminalMode};
+use windows::core::{GUID, HRESULT, PCSTR};
+use windows::s;
+use windows::Win32::Foundation::{BOOL, HINSTANCE};
+use windows::Win32::System::LibraryLoader::{GetProcAddress, LoadLibraryA};
+use windows::Win32::UI::WindowsAndMessaging::{LoadCursorW, SetCursor, IDC_ARROW};
+struct Dx12HookExample;
+
+impl Dx12HookExample {
+    fn new() -> Self {
+        hudhook::utils::alloc_console();
+        let config = ConfigBuilder::new()
+            .set_level_color(Level::Error, Some(Color::Magenta))
+            .set_level_color(Level::Trace, Some(Color::Green))
+            .set_target_level(LevelFilter::Error)
+            .build();
+        TermLogger::init(LevelFilter::Error, config, TerminalMode::Mixed, ColorChoice::Auto).ok();
+
+        Dx12HookExample
+    }
+}
+
+impl ImguiRenderLoop for Dx12HookExample {
+    fn should_block_messages(&self, _io: &mut imgui::Io) -> bool {
+        if _io.want_capture_mouse {
+            unsafe {
+                if let Ok(e) = LoadCursorW(None, IDC_ARROW) {
+                    SetCursor(e);
+                } else {
+                    SetCursor(None);
+                }
+            }
+            _io.mouse_draw_cursor = true;
+            return true;
+        } else {
+            _io.mouse_draw_cursor = false;
+            return false;
+        }
+    }
+
+    fn render(&mut self, ui: &mut imgui::Ui, _: &ImguiRenderLoopFlags) {
+        ui.window("Hello world").size([300.0, 110.0], Condition::FirstUseEver).build(|| {
```

Comment:
Using `ui.io_mut()` paired with `should_block_messages` here would be more appropriate for assessing whether to draw the cursor or not, and making the relevant modifications.